File: src/main/java/entities/gestionequipements/Equipements_details.java
package entities.gestionequipements;

public  class Equipements_details{

    private int id;
    private String name;
    private String description;
    private String duree_de_vie;
    private String etat;

    public Equipements_details() {
    }

    public Equipements_details(int id, String name, String description, String duree_de_vie, String etat) {
        this.id = id;
        this.name = name;
        this.description = description;
        this.duree_de_vie = duree_de_vie;
        this.etat = etat;
    }

    public Equipements_details(String name, String description, String duree_de_vie, String etat) {
        this.name = name;
        this.description = description;
        this.duree_de_vie = duree_de_vie;
        this.etat = etat;
    }

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public String getDuree_de_vie() {
        return duree_de_vie;
    }

    public void setDuree_de_vie(String duree_de_vie) {
        this.duree_de_vie = duree_de_vie;
    }

    public String getEtat() {
        return etat;
    }

    public void setEtat(String etat) {
        this.etat = etat;
    }

    @Override
    public String toString() {
        return "Equipements_details{" +
                "id=" + id +
                ", name='" + name + '\'' +
                ", description='" + description + '\'' +
                ", duree_de_vie='" + duree_de_vie + '\'' +
                ", etat='" + etat + '\'' +
                '}';
    }


}

File: webapp/src/GymPlus/src/Entity/EquipementsDetails.php
<?php

namespace App\Entity;

use Doctrine\ORM\Mapping as ORM;
use App\Repository\EquipementsDetailsRepository;
use Symfony\Component\Validator\Constraints as Assert;

#[ORM\Entity (repositoryClass: EquipementsDetailsRepository::class)]
class EquipementsDetails
{
    #[ORM\Id]
    #[ORM\GeneratedValue]
    #[ORM\Column]
    private ?int $id;

    #[ORM\Column]
    #[Assert\NotBlank(message: 'Please enter the equipment name.')]
    #[Assert\Length(min: 3, max: 255, minMessage: 'Name must be at least 3 characters long.', maxMessage: 'Name must be at most 255 characters long.')]
    #[Assert\Type(type: "string", message: 'Name must be a string.')]
    #[Assert\Regex(pattern: "/[a-zA-Z0-9]*/", message: 'Name must contain only letters and numbers.')]
    private ?string $name;

    #[ORM\Column]
    #[Assert\NotBlank(message: 'Please enter the equipment description.')]
    #[Assert\Length(min: 3, max: 255, minMessage: 'Description must be at least 3 characters long.', maxMessage: 'Description must be at most 255 characters long.')]
    #[Assert\Type(type: "string", message: 'Description must be a string.')]
    #[Assert\Regex(pattern: "/[a-zA-Z0-9]*/", message: 'Description must contain only letters and numbers.')]
    private ?string $description;

    #[ORM\Column]
    #[Assert\NotBlank(message: 'Please enter the equipment duree de vie.')]
    #[Assert\Length(min: 1, max: 255, minMessage: 'Duree de vie must be at least 1 character long.', maxMessage: 'Duree de vie must be at most 255 characters long.')]
    #[Assert\Type(type: "string", message: 'Duree de vie must be a string.')]
    #[Assert\Regex(pattern: "/[0-9]*/", message: 'Duree de vie must contain only numbers.')]
    private ?string $dureeDeVie;

    #[ORM\Column]
    #[Assert\NotBlank(message: 'Please enter the equipment etat.')]
    #[Assert\Length(min: 3, max: 255, minMessage: 'Etat must be at least 3 characters long.', maxMessage: 'Etat must be at most 255 characters long.')]
    #[Assert\Type(type: "string", message: 'Etat must be a string.')]
    #[Assert\Regex(pattern: "/[a-zA-Z0-9]*/", message: 'Etat must contain only letters and numbers.')]
    private ?string $etat;

    public function getId(): ?int
    {
        return $this->id;
    }

    public function getName(): ?string
    {
        return $this->name;
    }

    public function setName(?string $name): static
    {
        $this->name = $name;

        return $this;
    }

    public function getDescription(): ?string
    {
        return $this->description;
    }

    public function setDescription(?string $description): static
    {
        $this->description = $description;

        return $this;
    }

    public function getDureeDeVie(): ?string
    {
        return $this->dureeDeVie;
    }

    public function setDureeDeVie(?string $dureeDeVie): static
    {
        $this->dureeDeVie = $dureeDeVie;

        return $this;
    }

    public function getEtat(): ?string
    {
        return $this->etat;
    }

    public function setEtat(?string $etat): static
    {
        $this->etat = $etat;

        return $this;
    }


}
